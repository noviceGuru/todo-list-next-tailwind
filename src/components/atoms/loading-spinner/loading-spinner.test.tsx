import { render, screen } from "@testing-library/react"
import '@testing-library/jest-dom'

import LoadingSpinner from "./loading-spinner"

test('With no type, will have "animate-ping" class', () => {
	render(<LoadingSpinner />)
	const spinner = screen.getByText(/loading.../i)
	expect(spinner).toHaveClass('animate-ping')
	expect(spinner).toMatchSnapshot()
})

test('If passed a "pulse" type, will have "animate-pulse" class', () => {
	render(
		<LoadingSpinner
			type="pulse" />
	)
	const spinner = screen.getByText(/loading.../i)
	expect(spinner).toHaveClass('animate-pulse')
	expect(spinner).toMatchSnapshot()
})