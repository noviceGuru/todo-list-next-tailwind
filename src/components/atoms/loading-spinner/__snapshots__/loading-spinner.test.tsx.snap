// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If passed a "pulse" type, will have "animate-pulse" class 1`] = `
<span
  class="bg-indigo-300 self-center py-1 px-4 rounded-md animate-pulse"
>
  Loading...
</span>
`;

exports[`With no type, will have "animate-ping" class 1`] = `
<span
  class="bg-indigo-300 self-center py-1 px-4 rounded-md animate-ping"
>
  Loading...
</span>
`;
